### ISSUE TITLE
Adobe Accessibility Checker:  Appropriate nesting - Failed

### ISSUE THREAD

**Issue by kmagre** [josephwright/ltx-talk#111]
**Describe the issue**
When running the Adobe Accessibility Checker on a PDF generated by ltx-talk that includes one or more  sections defined by a \section command, the document fails the check:  Appropriate nesting.  When looking in the Tags panel, it looks like the `<H4>` header for the frametitle of the first frame in the section comes before the `<H1>` header for the \section command. 

**To Reproduce**
```latex
\DocumentMetadata{tagging=on}
\documentclass{ltx-talk}

  % Any preamble code goes here

\begin{document}

\section{Section 1}

\begin{frame}
\frametitle{Frame 1}
Content of Frame 1
\end{frame}

\begin{frame}
\frametitle{Frame 2}
Content of Frame 2
\end{frame}

\section{Section 2}

\begin{frame}
\frametitle{Frame 3}
Content of Frame 3
\end{frame}

\begin{frame}
\frametitle{Frame 4}
Content of Frame 4
\end{frame}
  
\end{document}
```

**Expected behavior**
I'm assuming that the `<H1>` tag from a \section command would come before the `<H4>` tag from the frametitle of each slide in that section.

**Comment by u-fischer** [comment]
@josephwright  Structures do not wait for shipout.  But with the LaTeX release `tag_tool:n` is deprecated because we are switching to tagging sockets, and while I didn't drop the command and the keys, it is quite possible that the key has not been properly adapted. I will take a look.

**Comment by josephwright** [comment]
@kmagre I'm integrated @u-fischer's changes - can you test with the 'burning edge' version of the class?

**Comment by u-fischer** [comment]
> [@kmagre](https://github.com/kmagre) I'm integrated [@u-fischer](https://github.com/u-fischer)'s changes - can you test with the 'burning edge' version of the class?

I do not think that it will solve the problem with the checker, as there is still an H4 directly after an H1 (or H2) and the checker won't like that. But you could offer an option to give a frame always "the next number" (this would then naturally mean that it can be sometimes H2 and sometimes H3 depending on if there is a \subsection or not).

**Comment by josephwright** [comment]
@u-fischer The H4 aspect is different - a discussion point - the nesting is a bug.

**Comment by josephwright** [comment]
We still don't really know if frame titles should be tagged as structure at all

**Comment by FrankMittelbach** [comment]
> However, as predicted, the Adobe Accessibility checker does still fail the check for Appropriate nesting.
> 
> In addition, the Adobe Accessibility checker now also fails Alternate Text -> Associated with content, but I don't know why.

@kmagre It would be good if we start collecting accessibility checker failures at https://github.com/latex3/tagging-project/discussions/categories/accessibility-checker-issues to build up a collection of them in a central place.  Some may be true  code issues (in which case they should and will be sorted out), but most are really failures in the checkers to correctly deal with PDF 2.0 and only but documenting them and confronting the vendors with their problems there is a chance to get to real accessibility in STEM documents, because 1.7 is just unsuitable when math  is part of the equation.

**Comment by u-fischer** [comment]
> In addition, the Adobe Accessibility checker now also fails Alternate Text -> Associated with content, but I don't know why.

Because I added an ActualText entry to the structures for `\section` and `\subsection` to improve their reading. These structures are empty and looks as if the adobe checker does not like that. But imho nothing in the spec says that this is not allowed ...

**Comment by teepeemm** [comment]
I assume FrankMittelbach meant to link to https://github.com/latex3/tagging-project/discussions/categories/issues-with-accessibility-checkers-and-other-at-software above.

**Comment by cfr42** [comment]
> I assume FrankMittelbach meant to link to https://github.com/latex3/tagging-project/discussions/categories/issues-with-accessibility-checkers-and-other-at-software above.

That doesn't actually seem to be somewhere for people to report problems with validation, though? I see guidelines for 'entries in this category', but I don't see any (place for) issues?

**Comment by teepeemm** [comment]
@cfr42 Clicking on the "New discussion" seems to create a new entry in that category.  I assume we don't want to create "Issues" because these aren't items that the LaTeX team needs to resolve, but we're instead compiling a database to which we'll point PDF vendors (and other interested parties).

**Comment by cfr42** [comment]
@teepeemm thanks. I shouldn't have used 'issue'. I meant I could only see @FrankMittelbach 's guidelines. I couldn't see any actual reports re. validators.

But probably this is just me. I'm not really familiar with how discussions work.

**Comment by FrankMittelbach** [comment]
> [@cfr42](https://github.com/cfr42) Clicking on the "New discussion" seems to create a new entry in that category. I assume we don't want to create "Issues" because these aren't items that the LaTeX team needs to resolve, but we're instead compiling a database to which we'll point PDF vendors (and other interested parties).

The idea of this category was/is that "issues" with individual checkers are recorded there as separate discussions, i.e., one per "issue"  with a particular  checker software. So far there is only one such entry (done very recently by @teepeemm ) but that is precisely how I thought this should work. The more we collect the better and that then gives us something to point vendors to.

**Comment by cfr42** [comment]
@teepeemm Thanks. I just assumed there would be extant issues if I looked in the right place, but, yes, I can see you report the issue I was thinking of ;).
